airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/version.py

```
@@ -41,15 +42,33 @@ def _should_run(self) -> bool:
 
     def _get_master_metadata(self, connector: Connector) -> Dict[str, Any] | None:
         """Get the metadata from the master branch or None if unable to retrieve."""
-        # TODO: test out if this works on the private airbyte-enterprise repo - consider using git-based approach
-        github_url_prefix = "https://raw.githubusercontent.com/airbytehq/airbyte/master/airbyte-integrations/connectors"
-        master_metadata_url = f"{github_url_prefix}/{connector.technical_name}/{consts.METADATA_FILE_NAME}"
-        response = requests.get(master_metadata_url)
+        cwd = Path.cwd().absolute()
+        repo_name = "airbyte-enterprise" if "airbyte-enterprise" in cwd.parts else "airbyte"
 
-        # New connectors will not have a metadata file in master
-        if not response.ok:
+        fetch_command = [
+            "gh",
+            "api",
+            f"repos/airbytehq/{repo_name}/contents/airbyte-integrations/connectors/{connector.technical_name}/{consts.METADATA_FILE_NAME}?ref=master",
+            "-H",
+            "Accept: application/vnd.github.v3.raw",
+        ]
+
+        try:
+            completed_process = subprocess.run(
+                fetch_command,
+                text=True,
+                check=False,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+
+            # New connectors will not have a metadata file in master
+            if completed_process.returncode != 0:
+                return None
+
+            return yaml.safe_load(completed_process.stdout)["data"]
+        except (subprocess.SubprocessError, yaml.YAMLError, KeyError):
```

(aside) it seems like we may not want to catch these exceptions here and instead let them bubble up. If this method returns `None` it seems like we assume the connector doesn't exist on master and set the "master version" to `0.0.0`. Probably makes sense instead to fail the check.